### PR TITLE
[in_app_purchase_android] Bump com.android.billingclient:billing from 5.2.0 to 6.0.0.

### PR DESCRIPTION
**Pull request by gmackall**
Takeover of https://github.com/flutter/packages/pull/3988/

Conflicts had built up and it required some additional changes so I figured it would be easier to just make my own new branch to fix.

Changes from last PR
1. I made the particular ProrationMode enum value we were using a constant and put the deprecation suppression on that constant.
2. I made a helper to wrap the other deprecated method (`setReplaceProrationMode`).
3. Added todos with links to the issue for replacing the deprecated classes/methods.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/in_app_purchase/in_app_purchase_android/android/src/main/java/io/flutter/plugins/inapppurchase/MethodCallHandlerImpl.java [hunk 1]
```diff
@@ -322,15 +329,24 @@ private void launchBillingFlow(
         BillingFlowParams.SubscriptionUpdateParams.newBuilder();
     if (oldProduct != null && !oldProduct.isEmpty() && purchaseToken != null) {
       subscriptionUpdateParamsBuilder.setOldPurchaseToken(purchaseToken);
-      // The proration mode value has to match one of the following declared in
-      // https://developer.android.com/reference/com/android/billingclient/api/BillingFlowParams.ProrationMode
-      subscriptionUpdateParamsBuilder.setReplaceProrationMode(prorationMode);
+      //Set the prorationMode using a helper to minimize impact of deprecation suppression.
```
Comment (stuartmorgan) [inline packages/in_app_purchase/in_app_purchase_android/android/src/main/java/io/flutter/plugins/inapppurchase/MethodCallHandlerImpl.java:332]: Nit: missing space after `//`

File: packages/in_app_purchase/in_app_purchase_android/android/src/test/java/io/flutter/plugins/inapppurchase/MethodCallHandlerTest.java [hunk 2]
```diff
@@ -417,6 +417,7 @@ public void launchBillingFlow_ok_AccountId() {
   }
 
   @Test
+  @SuppressWarnings(value = "deprecation")
```
Comment (stuartmorgan) [inline packages/in_app_purchase/in_app_purchase_android/android/src/test/java/io/flutter/plugins/inapppurchase/MethodCallHandlerTest.java:420]: Could you annotate these with the TODO comment linking to the issue, so we know why they are here and when we can clean them up?

**Review by stuartmorgan — APPROVED** [review]
LGTM with nits, thanks for downscoping the warning suppression!

File: packages/in_app_purchase/in_app_purchase_android/android/src/test/java/io/flutter/plugins/inapppurchase/MethodCallHandlerTest.java (hunk 2 above)
Comment (gmackall) [reply packages/in_app_purchase/in_app_purchase_android/android/src/test/java/io/flutter/plugins/inapppurchase/MethodCallHandlerTest.java:423]: Added!

**Pull request merged**